### ISSUE TITLE
Formatting GLSL not supported, even though it should be able to format GLSL

### ISSUE THREAD

**Issue by FrostKiwi** [microsoft/vscode-cpptools#9902]
### Environment

- OS and version: Windows_NT x64 10.0.19044
- VS Code: 1.71.2 (user setup)
- C/C++ extension: v1.12.4
- OS and version of remote machine (if applicable):

### Bug Summary and Steps to Reproduce

Posting issue here, as suggested by: https://github.com/microsoft/vscode/issues/161209#issuecomment-1251914962

Issuing the format document command on a GLSL file with .frag as the extension in case of the this fragment shader file, results in the `There is no formatter for 'glsl' files installed.` error message. Which is correct, I do not have one installed specifically for GLSL shaders.

![image](https://user-images.githubusercontent.com/60887273/190934773-d8f2ac01-a225-4aa0-9428-a7c2f4a52dfc.png)

I would like to use the default formatter for C files, as GLSL is formatted identically to C. So I set the following in my settings .json
```
    "[glsl]": {
        "editor.defaultFormatter": "ms-vscode.cpptools"
    },
```
This does not change the error message, which persists. 

I presume this is because ms-vscode.cpptools is not specifically registered to work with glsl files, although the formatter should be more than capable of formatting GLSL files. It would be excellent, if I could chose the C formatter for GLSL files.

Does this issue occur when all extensions are disabled?: Yes

<details><summary>Detailed version Info</summary>

```
Version: 1.71.2 (user setup)
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:03:37.738Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No
```
</details>

Steps to Reproduce:

1. Set "editor.defaultFormatter": "ms-vscode.cpptools"
2. Format a glsl file

**Comment by sean-mcmanus** [comment]
Do you intend to format with vcFormat or clangFormat? If you set `files.associations: { "*.frag": "c" }` that should be a workaround, but it would treat the file as C for all language service features. I'm not sure formatting would work without issues. We'd also have to disable non-formatting features like IntelliSense.

**Comment by FrostKiwi** [comment]
> set `files.associations: { "*.frag": "c" }` that should be a workaround

Indeed, that works and formats the file correctly, but as you already guessed, IntelliSense flips out with errors:
![image](https://user-images.githubusercontent.com/60887273/191420624-4e14acbf-b4c3-4c0b-9aed-8ae6e981fe4f.png)
I see no formatting issues with multiple files across GLSL vertex and fragment shaders. I presume the same to be true for DirectX HLSL shaders.

I wonder if I can setup a function,, that temporarily switches to files.associations: c, format file, and switch back to the original association as an extra dirty workaround. Bind it to the default Shift+Alt+F with .frag files as the conditional.
> intend to format with vcFormat or clangFormat?

I never set it to clangFormat, so I presume the default(?) vcFormat is being used.

> We'd also have to disable non-formatting features like IntelliSense.

I kinda presumed this "formatter" thingy is an extra component divorced from linting and stuff, but clearly this is a bit more complicated than I would have guessed...

**Comment by sean-mcmanus** [comment]
You could try setting the C_Cpp.errorSquiggles to "Disabled". You may need to disable other language service features too. However, that would only work if you also don't want those in actual C/C++ files, unless possibly if you used a multi-root workspace with one workspace folder configured for GLSL formatting and another for C/C++.

The default is clangFormat.

I'm not sure if switching file associations back and forth would work.

The reason the formatting is embedded with the other language service features is due to our shared internal document object which is used by formatting and IntelliSense.

**Comment by FrostKiwi** [comment]
>The default is clangFormat.

Ohh, good to know.
> You could try setting the C_Cpp.errorSquiggles to "Disabled"

I must remain in glsl mode to get the glsl linting via https://marketplace.visualstudio.com/items?itemName=dtoplak.vscode-glsllint .
> I'm not sure if switching file associations back and forth would work.

It does! Hahaha, this is so ugly, but man is it ever better than installing an extra formatter:
``` json
{
    "key": "ctrl+alt+f",
    "command": "extension.multiCommand.execute",
    "args": {
        "sequence": [
            { "command": "commands.setEditorLanguage", "args": "c"},
            { "command": "editor.action.formatDocument"},
            { "command": "commands.setEditorLanguage", "args": "glsl"}
        ]
}
 
```
Using https://marketplace.visualstudio.com/items?itemName=usernamehw.commands to change the language mode and https://marketplace.visualstudio.com/items?itemName=ryuta46.multi-command to execute multiple commands with one keybind.
It works and formats correctly in roughly half a second, so this is perfect.

**Comment by lessquo** [comment]
You can use https://marketplace.visualstudio.com/items?itemName=xaver.clang-format instead, but you need to install clang-format separately for this extension

**Comment by FrostKiwi** [comment]
> You can use https://marketplace.visualstudio.com/items?itemName=xaver.clang-format instead, but you need to install clang-format separately for this extension

Sounds great, will try it out! I step up from from using.
```json
{
    "key": "ctrl+alt+f",
    "command": "extension.multiCommand.execute",
    "args": {
        "sequence": [
            { "command": "commands.setEditorLanguage", "args": "c"},
            { "command": "editor.action.formatDocument"},
            { "command": "commands.setEditorLanguage", "args": "glsl"}
        ]
}
 
```
Installing an extra dependency isn't that bad for me, since I'm installing https://github.com/KhronosGroup/glslang + https://marketplace.visualstudio.com/items?itemName=dtoplak.vscode-glsllint anyways.

**Comment by bobbrow** [comment]
> You can use https://marketplace.visualstudio.com/items?itemName=xaver.clang-format instead, but you need to install clang-format separately for this extension

I'm not sure I understand what you mean. We bundle clang-format with this extension and it's the default formatter.

**Comment by bobbrow** [comment]
> > You can use https://marketplace.visualstudio.com/items?itemName=xaver.clang-format instead, but you need to install clang-format separately for this extension
> 
> I'm not sure I understand what you mean. We bundle clang-format with this extension and it's the default formatter.

Oh, are you saying that you need to install clang-format yourself if you use the other extension?

**Comment by FrostKiwi** [comment]
> We bundle clang-format with this extension and it's the default formatter.

Ohh, you are correct, I confused what is included and what isn't.

**Comment by lessquo** [comment]
> > You can use https://marketplace.visualstudio.com/items?itemName=xaver.clang-format instead, but you need to install clang-format separately for this extension
> 
> I'm not sure I understand what you mean. We bundle clang-format with this extension and it's the default formatter.

Yes, but this issue is saying that the `ms-vscode.cpptools` extension cannot format GLSL files. So, I suggested an alternative. With the `xaver.clang-format` extension, and while the clang-format is installed on your machine, formatting GLSL files works with this simple setting:

```json
"[glsl]": {
  "editor.defaultFormatter": "xaver.clang-format"
},
```
